### DISCUSSION TITLE
Vague "Could not extract" error with autoReplaceStringTemplate

### DISCUSSION THREAD

**Discussion by uhthomas** [renovatebot/renovate#24021] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

I'm trying to rewrite some generic strings with Renovate. For our example, I just want Renovate to replace `golang.org/*` with `example.com/*`. To do this, I'm lying to Renovate and providing a static value to `currentVersion`. It fails with a very vague message and I'm not sure exactly what to do to achieve what I want or fix the configuration. I also think it would be nice to improve the error message.

```
DEBUG: Could not extract tools/tools.go (branch="renovate/example.com-x-tools-go-analysis-1.x")
WARN: Error updating branch: update failure (branch="renovate/example.com-x-tools-go-analysis-1.x")
```

link to repository: [https://github.com/uhthomas/renovate24021](https://github.com/uhthomas/renovate24021)

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by rarkins** [comment, marked as answer]
PRs welcome

**Reply by uhthomas** [reply]
@rarkins Any idea what could be causing the issue here?

**Reply by rarkins** [reply]
If you offer completely flexibility, it will always be possible for users to have faulty/wrong/non-functioning config. Sometimes it's like if someone expected the regex engine to explain why it didn't match.

The debug line itself is https://github.com/renovatebot/renovate/blob/0613c2472958884aa7ff33b9e80511f0acb63bf7/lib/workers/repository/update/branch/auto-replace.ts#L27-L31

It implies that the package file could not be parsed after the autoreplace

**Reply by uhthomas** [reply]
I'm not really sure how. I tried with a more stripped down and explicit configuration:

```json
"regexManagers": [{
  "fileMatch": ["\\.go$"],
  "matchStrings": ["\"(?<depName>golang.org/x/tools/go/analysis)\""],
  "depNameTemplate": "example.com/x/tools/go/analysis",
  "autoReplaceStringTemplate": "\"example.com/x/tools/go/analysis\"",
  "datasourceTemplate": "custom.fake",
  "currentValueTemplate": "1.0.0"
}],
```

and these variations for `autoReplaceStringTemplate`:

- `"example.com/x/tools/go/analysis"`
- `"\"example.com/x/tools/go/analysis\""`
- `"golang.org/x/tools/go/analysis"`
- `"\"golang.org/x/tools/go/analysis\""`

Which all fail with the exact same error.

**Reply by rarkins** [reply]
I suggest you try instead the replacementX configuration options which are designed for that purpose, e.g. https://docs.renovatebot.com/configuration-options/#replacementname

**Reply by uhthomas** [reply]
Thanks! I would love to, but the documentation explicitly states that both `gomod` and `regex` are unsupported.

<img width="352" alt="Screenshot 2023-08-22 at 16 07 03" src="https://github.com/renovatebot/renovate/assets/9749173/7410af8e-16e0-495f-b98a-0fa43c7cc141">

**Reply by uhthomas** [reply]
fwiw I was trying to replicate the example from https://docs.renovatebot.com/configuration-options/#autoreplacestringtemplate

**Reply by HonkingGoose** [reply]
> Thanks! I would love to, but the documentation explicitly states that both `gomod` and `regex` are unsupported.

`@jamiemagie` (maintainer) updated the list of managers that do not support replacement in this issue in January of this year:

- #14149

Looks like the `gomod` and `regex` managers _still_ don't support replacement. I'll keep the docs as they are now.

**Reply by uhthomas** [reply]
@HonkingGoose Any ideas on what's actually going on with my configuration? It's a bit unorthodox, but it should work?

**Reply by HonkingGoose** [reply]
> Any ideas on what's actually going on with my configuration? It's a bit unorthodox, but it should work?

I don't know enough to help you. Maybe the better logging from the PR below will help you figure out what's wrong. You'll have to wait for the PR to be merged _and_ the Mend Renovate app to be updated.

> PRs welcome

This PR is adding better log messages when autoreplace fails:

- #24042
